backend/src/main/java/de/nordakademie/iaa/noodle/services/UserService.java

```diff
@@ -0,0 +1,44 @@
+package de.nordakademie.iaa.noodle.services;
+
+import de.nordakademie.iaa.noodle.dao.UserRepository;
+import de.nordakademie.iaa.noodle.filter.NoodleException;
+import de.nordakademie.iaa.noodle.model.User;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.dao.DataIntegrityViolationException;
+import org.springframework.security.crypto.bcrypt.BCryptPasswordEncoder;
+import org.springframework.security.crypto.password.PasswordEncoder;
+import org.springframework.stereotype.Component;
+
+import java.util.HashSet;
+import java.util.Optional;
+
+import static de.nordakademie.iaa.noodle.config.SecurityConstants.HASH_PEPPER;
+
+@Component
+public class UserService {
+    private final UserRepository userRepository;
+
+    @Autowired
+    public UserService(UserRepository userRepository) {
+        this.userRepository = userRepository;
+    }
+
+    public Optional<User> getUserByEMail(String email) {
+        return Optional.ofNullable(userRepository.findByEmail(email));
+    }
+
+    public Optional<User> getUserByUserID(long userID) {
+        return Optional.ofNullable(userRepository.findByid(userID));
+    }
+
```

Comment:
I'm not entirely convinced these are totally necessary. Also, we usually use `Long`

Reply:
Changed to Long, left the Optionals in since the handling is better in consuming classes